pos_ask_vat/static/src/js/Misc/NumberBuffer.js

```diff
@@ -0,0 +1,95 @@
+
+    var NumberBuffer = require("point_of_sale.NumberBuffer");
+    const {Gui} = require("point_of_sale.Gui");
+
+    const _updateBuffer = (input) => {
+        const self = NumberBuffer;
+        const isEmpty = (val) => {
+            return val === "" || val === null;
+        };
+        if (input === undefined || input === null) return;
+        const isFirstInput = isEmpty(self.state.buffer);
+        if (input === "," || input === ".") {
+            if (isFirstInput) {
+                self.state.buffer = "0" + self.decimalPoint;
+            } else if (!self.state.buffer.length || self.state.buffer === "-") {
+                self.state.buffer += "0" + self.decimalPoint;
+            } else if (self.state.buffer.indexOf(self.decimalPoint) < 0) {
+                self.state.buffer += self.decimalPoint;
+            }
+        } else if (input === "Delete") {
+            if (self.isReset) {
+                self.state.buffer = "";
+                self.isReset = false;
+                return;
+            }
+            self.state.buffer = isEmpty(self.state.buffer) ? null : "";
+        } else if (input === "Backspace") {
+            if (self.isReset) {
+                self.state.buffer = "";
+                self.isReset = false;
+                return;
+            }
+            const buffer = self.state.buffer;
+            if (isEmpty(buffer)) {
+                self.state.buffer = null;
+            } else {
+                const nCharToRemove =
+                    buffer[buffer.length - 1] === self.decimalPoint ? 2 : 1;
+                self.state.buffer = buffer.substring(0, buffer.length - nCharToRemove);
+            }
+        } else if (input === "+") {
+            if (self.state.buffer[0] === "-") {
+                self.state.buffer = self.state.buffer.substring(
+                    1,
+                    self.state.buffer.length
+                );
+            }
+        } else if (input === "-") {
+            if (isFirstInput) {
+                self.state.buffer = "-0";
+            } else if (self.state.buffer[0] === "-") {
+                self.state.buffer = self.state.buffer.substring(
+                    1,
+                    self.state.buffer.length
+                );
+            } else {
+                self.state.buffer = "-" + self.state.buffer;
+            }
+        } else if (input[0] === "+" && !isNaN(parseFloat(input))) {
+            // When input is like '+10', '+50', etc
+            const inputValue = parseFloat(input.slice(1));
+            const currentBufferValue = self.state.buffer
+                ? parseFloat(self.state.buffer)
+                : 0;
+            self.state.buffer = self.component.env.pos.formatFixed(
+                inputValue + currentBufferValue
+            );
+        } else if (!isNaN(parseInt(input, 10))) {
+            if (isFirstInput) {
+                self.state.buffer = String(input);
+            } else if (self.state.buffer.length > 13) {
```

Comment:
max length is 12 in the original code. Did you change it specifically for Brazil? Is this generic?